mlflow/tracking/_default_experiment/databricks_job_context.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from mlflow.tracking._default_experiment.abstract_context import DefaultExperimentProvider
+from mlflow.utils import databricks_utils
+from mlflow.entities import SourceType
+from mlflow.tracking.client import MlflowClient
+from mlflow.utils.mlflow_tags import (
+    MLFLOW_DATABRICKS_JOB_TYPE_INFO,
+    MLFLOW_EXPERIMENT_SOURCE_TYPE,
+    MLFLOW_EXPERIMENT_SOURCE_ID,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class DatabricksJobExperimentContext(DefaultExperimentProvider):
+    def in_context(self):
+        return (
+            databricks_utils.is_in_databricks_job()
+            and databricks_utils.get_job_type_info() == "NORMAL"
+        )
+
+    def get_experiment_id(self):
+        job_id = databricks_utils.get_job_id()
+        tags = {}
+        tags[MLFLOW_DATABRICKS_JOB_TYPE_INFO] = databricks_utils.get_job_type_info()
```

Comment:
Can we add inline comments to https://github.com/mlflow/mlflow/blob/3c434246f73b41835194f3e6aa1f9be8488bcd56/mlflow/utils/mlflow_tags.py#L44-L45 explaining the content difference between these two tags?